### ISSUE TITLE
Loss is not decreasing

### ISSUE THREAD

**Issue by MananAmin** [minimaxir/gpt-2-simple#207]
file_name = "/kaggle/input/quotes-500k/train.txt"
sess = gpt2.start_tf_sess()
gpt2.finetune(sess,
              file_name,
              model_name=model_name,        #"124M"
              steps=100)   # steps is max number of training steps

This is the code I used for fine-tuning but even after 100 iterations.. loss isn't changing.
Is it because of too large dataset or because of hyper-parameters?

**Comment by WAUthethird** [comment]
100 steps is pretty low, I'd recommend increasing that.
And no, your dataset is decently small, but that shouldn't affect the loss too much.

**Comment by MananAmin** [comment]
Thanks  
I will try it for 500 epochs/iterations..
I at least expecting that gpt-2 would give better results than lstm. 
In generation, I got repetitive sentences.. that's I think usual behaviour of gpt-2.

**Comment by WAUthethird** [comment]
You'll get better results if you use a larger model by the way, keep that in mind if you get poor results

**Comment by MananAmin** [comment]
btw, Can you please glance at the loss behavior of this training..
Is it a normal behavior of GPT-2 while training?

[Notebook](https://www.kaggle.com/manann/generating-quotes-using-gpt-2-language-model?scriptVersionId=33014654)
